src/shared/transitions.js

```diff
@@ -4,37 +4,31 @@ export function linear(t) {
 	return t;
 }
 
-export function generateKeyframes(
+export function generateRule(
 	a,
 	b,
 	delta,
 	duration,
 	ease,
-	fn,
-	node,
-	style
+	fn
 ) {
-	var id = '__svelte' + ~~(Math.random() * 1e9); // TODO make this more robust
-	var keyframes = '@keyframes ' + id + '{\n';
+	var keyframes = '{\n';
 
 	for (var p = 0; p <= 1; p += 16.666 / duration) {
 		var t = a + delta * ease(p);
 		keyframes += p * 100 + '%{' + fn(t) + '}\n';
 	}
 
-	keyframes += '100% {' + fn(b) + '}\n}';
-	style.textContent += keyframes;
+	return keyframes + '100% {' + fn(b) + '}\n}';
+}
 
-	document.head.appendChild(style);
+// https://github.com/darkskyapp/string-hash/blob/master/index.js
+export function hash(str) {
+	var hash = 5381;
+	var i = str.length;
 
-	node.style.animation = (node.style.animation || '')
-		.split(',')
-		.filter(function(anim) {
-			// when introing, discard old animations if there are any
-			return anim && (delta < 0 || !/__svelte/.test(anim));
-		})
-		.concat(id + ' ' + duration + 'ms linear 1 forwards')
-		.join(', ');
+	while (i--) hash = (hash * 33) ^ str.charCodeAt(i);
```

Comment:
Wait, how did I actually think 33 is a prime number... anyways 33 is bad because it's also not a prime number.

This is from the book _Effective Java 2nd Edition_ which is a very good book:

> 
> The value 31 was chosen because it's an odd prime. If it were even and multiplication overflowed, information would be lost, as multiplication by 2 is equivalent to shifting. The advantage of using a prime is less clear, but it is traditional.
> 
> A nice property of 31 is that the multiplication can be replaced by a shift and subtraction for better performance:
> 
>  31 * i == (i << 5) - i
> 
> Modern VMs do this sort of optimization automatically.